src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/commands.py

```diff
@@ -17,205 +15,139 @@
+    )
+
+    virtual_network_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.devtestlabs.operations.virtual_networks_operations#VirtualNetworksOperations.{}',
+        client_factory=get_devtestlabs_virtual_network_operation
+    )
+
+    formula_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.devtestlabs.operations.formulas_operations#FormulasOperations.{}',
+        client_factory=get_devtestlabs_formula_operation
+    )
+
+    secret_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.devtestlabs.operations.secrets_operations#.SecretsOperations{}',
+        client_factory=get_devtestlabs_secret_operation,
+        validator=validate_user_name
+    )
+
+    environment_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.devtestlabs.operations.environments_operations#EnvironmentsOperations.{}',
+        client_factory=get_devtestlabs_environment_operation,
+        validator=validate_user_name
+    )
+
+    arm_template_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.devtestlabs.operations.arm_templates_operations#ArmTemplatesOperations.{}',
+        client_factory=get_devtestlabs_arm_template_operation
+    )
 
     # Virtual Machine Operations Commands
-    virtual_machine_operations = create_service_adapter(
-        mgmt_operations_path.format('virtual_machines_operations'),
-        'VirtualMachinesOperations')
-
-    with ServiceGroup(__name__, get_devtestlabs_virtual_machine_operation,
-                      virtual_machine_operations) as s:
-        with s.group('lab vm') as c:
-            c.command('show', 'get', table_transformer=transform_vm)
-            c.command('delete', 'delete')
-            c.command('start', 'start')
-            c.command('stop', 'stop')
-            c.command('apply-artifacts', 'apply_artifacts')
-
-    # Virtual Machine Operations Custom Commands
-    with ServiceGroup(__name__, get_devtestlabs_virtual_machine_operation,
-                      custom_operations) as s:
-        with s.group('lab vm') as c:
-            c.command('list', 'list_vm', table_transformer=transform_vm_list)
-            c.command('claim', 'claim_vm')
-
-    # Lab Operations Custom Commands
-    with ServiceGroup(__name__, get_devtestlabs_lab_operation,
-                      custom_operations) as s:
-        with s.group('lab vm') as c:
-            c.command('create', 'create_lab_vm')
-
-    lab_operations = create_service_adapter(mgmt_operations_path.format('labs_operations'),
-                                            'LabsOperations')
+    with self.command_group('lab vm', virtual_machine_operations) as g:
+        g.command('show', 'get', table_transformer=transform_vm)
+        g.command('delete', 'delete')
+        g.command('start', 'start')
+        g.command('stop', 'stop')
+        g.command('apply-artifacts', 'apply_artifacts')
+        g.custom_command('list', 'list_vm', client_factory=get_devtestlabs_virtual_machine_operation,
```

Comment:
This *should* also work:
`with self.command_group('lab vm', virtual_machine_operations, client_factory=get_devtestlabs_virtual_machine_operations) as g:`

If it doesn't... I'll make it work!

Reply:
This will only work if the command_type does not have the kwarg('client_factory') specified. 
Right now any kwargs in the <custom_>command_type will override the ones in the group kwargs:
https://github.com/Azure/azure-cli/blob/KnackConversion/src/azure-cli-core/azure/cli/core/commands/__init__.py#L689-L694

Reply:
@tjprescott once the above issue is fixed, i.e: 

(command kwargs) overrides => (group kwargs) overrides => (command-type kwargs)

We can fix https://github.com/Azure/azure-cli/issues/5072
By just saying, `kwargs.get('client_factory')` in `_cli_command()`, instead of looking into the command_type again.